### DISCUSSION TITLE
Comfyui and diffusers can not get consistent result

### DISCUSSION THREAD

**Discussion by jyangliu** [huggingface/diffusers#7628] [Q&A]
Recently, I want to migrate comfyui workflows to huggingface code. First I test a simple examples from [diffusers DOC](https://huggingface.co/docs/diffusers/using-diffusers/ip_adapter), code is:
`from diffusers import AutoPipelineForText2Image
from diffusers.utils import load_image
import torch

pipeline = AutoPipelineForText2Image.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16).to("cuda")
pipeline.load_ip_adapter("h94/IP-Adapter", subfolder="sdxl_models", weight_name="ip-adapter_sdxl.bin")
pipeline.set_ip_adapter_scale(0.6)
image = load_image("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/ip_adapter_diner.png")
generator = torch.Generator(device="cpu").manual_seed(0)
images = pipeline(
    prompt="a polar bear sitting in a chair drinking a milkshake",
    ip_adapter_image=image,
    negative_prompt="deformed, ugly, wrong proportion, low res, bad anatomy, worst quality, low quality",
    num_inference_steps=100,
    generator=generator,
).images
images[0]`

My system is :

- Win10 + Torch2.0.0 + diffusers0.27.2

In my compute, I run the code, and i got a result like this:
![text2img](https://github.com/huggingface/diffusers/assets/44828355/443e6224-b052-455a-a727-1c9b1994c718)
which is different from official reslut, official result like this:
![ip_adapter_diner_2](https://github.com/huggingface/diffusers/assets/44828355/54af9e6a-aa0d-4338-9505-23944815ad7a)

Then I build a workflow with comfyui, and I got a result like this:
![IPAdapter_00554_](https://github.com/huggingface/diffusers/assets/44828355/438350e7-1cbc-4653-a05d-bdf95884623e)

I got three different result, and i don't know what wrong with that. My question is:

1. Why i got different result when i use the same code as official.
2. Is there anyone know how to convert comfyui workflow to diffusers code?

By the way, comfyui workflow is:
[workflow.json](https://github.com/huggingface/diffusers/files/14928420/workflow.json)

**Comment by asomoza** [comment]
Hi, it is practically impossible to get the same results. ComfyUI uses `sgm` and diffusers has its own implementation, even the official has a difference with comfyui and they use the same `sgm` library.

There's multiple factors that affect this, you can read some of them in this issue: https://github.com/huggingface/diffusers/issues/7432

If you're having problems with converting workflows to `diffusers` I can help you with that, I see that you already have done the one with IP Adapter. Just tell me if you have trouble converting another workflow or a specific node and I'll help you.

**Reply by jyangliu** [reply]
Thank you for your reply! I get the idea why ComfyUI and diffusers get different results. The first question, I used the same code as official DOC, Why I got a different result?

**Reply by asomoza** [reply]
Same reasons; different hardware, library versions, OS, etc. too many variables to know what exactly is making the result different. The docs and the guides are just a reference for you, maybe you'll get the same results or not, what is important is that you can reproduce the same result in your environment.

For instance I get the same result as you which is rare for me as I always get different results than the people that post here.

![float16](https://github.com/huggingface/diffusers/assets/5442875/4ce51938-5770-4a2f-b7bd-a456a7a73cb4)
